torchvision/datasets/sun397.py

```diff
@@ -0,0 +1,122 @@
+        parition (string, integer, optional): A valid partition can be an integer from 1 to 10 or ``"all"``
+            for the entire dataset.
+        transform (callable, optional): A function/transform that  takes in an PIL image and returns a transformed
+            version. E.g, ``transforms.RandomCrop``.
+        target_transform (callable, optional): A function/transform that takes in the target and transforms it.
+    """
+
+    _URL = "http://vision.princeton.edu/projects/2010/SUN/SUN397.tar.gz"
+    _FILENAME = "SUN397.tar.gz"
+    _MD5 = "8ca2778205c41d23104230ba66911c7a"
+    _PARTITIONS_URL = "https://vision.princeton.edu/projects/2010/SUN/download/Partitions.zip"
+    _PARTITIONS_FILENAME = "Partitions.zip"
+
+    def __init__(
+        self,
+        root: str,
+        split: str = "train",
+        partition: Union[int,str] = 1,
+        download: bool = True,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+    ) -> None:
+        super().__init__(root, transform=transform, target_transform=target_transform)
+        self.split = split
+        self.partition = partition
+        self.data_dir = Path(self.root) / "SUN397"
+
+        if download:
+            self._download()
+
+        if not self._check_exists():
+            raise RuntimeError("Dataset not found. You can use download=True to download it")
+        
+        self._labels = []
+        self._image_files = []
+        with open(self.data_dir / f"ClassName.txt", "r") as f:
+            classes = f.read().splitlines()
+            
+        for idx,c in enumerate(classes):
+            classes[idx] = c[3:]
+
+        self.classes = classes
+        self.class_to_idx = dict(zip(self.classes, range(len(self.classes))))
+        
+        if isinstance(self.partition,int):
+            if self.partition<0 or self.partition>10:
+                raise RuntimeError("Enter a valid integer partition from 1 to 10 or \"all\" ")
+            
+            splitname = "Training" if self.split is "train" else "Testing"
+            zero = "0" if self.partition<10 else ""
+            
+            with open(self.data_dir / f"{splitname}_{zero}{self.partition}.txt", "r") as f:
+                pathlist = f.read().splitlines()
+                
+            for p in pathlist:
+                self._labels.append(self.class_to_idx[p[3:-25]])
+                self._image_files.append(self.data_dir.joinpath(*p.split("/")))
+            
+        else:
+            if self.partition is not "all":
+                raise RuntimeError("Enter a valid integer partition from 1 to 10 or \"all\" ")
+            else:
+                for path, _, files in os.walk(self.data_dir):
+                    for file in files:
+                        if(file[:3]=="sun"):
+                            self._image_files.append(Path(path)/file)
+                            self._labels.append(Path(path).relative_to(self.data_dir).as_posix()[2:])
```

Comment:
The integer partition from 1-10 have files associated with them but, "all" doesn't have a file, so I manually browse the class directories and prepare a list out of it. 
That's why I think they cannot be merged. is it correct ?

Reply:
Assuming we have either an integer `partition` or `"all"`, you could do something like this:

```python
if partition == "all":
    partitions = list(range(1, 11))
else:
    partitions = [partition]

for partition in partitions:
    with open(self.data_dir / f"{self.split.title()}ing_{partition:02d}.txt", "r") as f:
        ...
```

This would also allow us to accept a collection of integers (or strings) for the `partition` parameter.

Reply:
Sorry I didn't mention it clearly before, but, the "partitions" are just a random subset of entire dataset consisting of 50 images per class(they aren't disjoint, all are random). So adding all of them will lead to repeated images.

Reply:
Oh, so `partition="all"` does not mean "all partitions", but rather "all images"? In that case, I suggest we change the value. What about `partition=None` to mean "don't give me any partitions, but rather all images"?

Reply:
Yes seems logical. I will make the change.